test_util/helpers.py

```diff
@@ -50,74 +49,50 @@ def authenticate(self, cluster):
 
 class ApiClient:
 
-    def __init__(self, cluster, user, api_base, default_headers=None, ca_cert_path=None):
-        self.cluster = cluster
-        self.user = user
+    def __init__(self, default_host_url, api_base, default_headers=None, ca_cert_path=None):
+        self.default_host_url = default_host_url
         self.api_base = api_base
         if default_headers is None:
             default_headers = dict()
         self.default_headers = default_headers
         self.ca_cert_path = ca_cert_path
 
-    def api_request(self, method, path, node=None, port=None, **kwargs):
+    @classmethod
+    def from_cluster(cls, cluster, api_base, default_headers=None):
+        if default_headers is None:
+            default_headers = {}
+        if cluster.auth_enabled and cluster.web_auth_default_user:
+            default_headers.update(cluster.web_auth_default_user.auth_header)
+        return cls(
+            default_host_url=cluster.dcos_uri,
+            api_base=api_base,
+            ca_cert_path=cluster.ca_cert_path,
+            default_headers=default_headers)
+
+    def api_request(self, method, path, host_url=None, **kwargs):
         """
         Makes a request with default headers + user auth headers if necessary
-        If DCOS_CA_CERT_PATH is set in environment, this method will pass it to requests
+        If self.ca_cert_path is set, this method will pass it to requests
         Args:
             method: (str) name of method for requests.request
-            node: see get_url()
             path: see get_url()
-            port: see get_url()
+            host_url: override the client's default host URL
             **kwargs: any optional arguments to be passed to requests.request
         """
         headers = copy.copy(self.default_headers)
-        if self.cluster.auth_enabled and self.cluster.web_auth_default_user:
-            headers.update(self.cluster.web_auth_default_user.auth_header)
 
-        if self.ca_cert_path:
+        if self.ca_cert_path and 'verify' not in kwargs:
```

Comment:
this allows people to make requests to ApiClient with verify=False or verify=my_other_custom_ca. Otherwise, the verify kwarg is always overridden, which makes for a bad requests wrapper